src/main/java/net/testudobank/MvcController.java

```diff
@@ -395,10 +404,18 @@ public String submitDispute(@ModelAttribute("user") User user) {
       // If reversing a deposit puts customer back in overdraft
       if (reversalAmountInPennies > userBalanceInPennies){
         // check if the reversed deposit helped pay off overdraft balance
-        List<Map<String,Object>> overdraftLogs = TestudoBankRepository.getOverdraftLogs(jdbcTemplate, userID, (String)logToReverse.get("Timestamp"));
+        String dateTime = SQL_DATETIME_FORMATTER.format(convertLocalDateTime((LocalDateTime)logToReverse.get("Timestamp")));
+        List<Map<String,Object>> overdraftLogs = TestudoBankRepository.getOverdraftLogs(jdbcTemplate, userID, dateTime);
+
+        // fetch updated overdraft balance and correct interest rate
+        double updatedOverdraftBalanceInPennies = TestudoBankRepository.getCustomerOverdraftBalanceInPennies(jdbcTemplate, userID);
+        double newOverdraftBalanceInPennies = updatedOverdraftBalanceInPennies / 1.02;
```

Comment:
Add a comment above this line saying `// reverse extra application of interest rate since customer was already in overdraft`.